invenio_vocabularies/services/custom_fields/vocabulary.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Custom Fields for InvenioRDM."""
+
+from invenio_access.permissions import system_identity
+from invenio_records_resources.records.systemfields import PIDListRelation, PIDRelation
+from invenio_records_resources.services.custom_fields.base import BaseCF
+from marshmallow import fields
+
+from ...proxies import current_service
+from ...resources.serializer import VocabularyL10NItemSchema
+from ...services.schema import VocabularyRelationSchema
+
+
+class VocabularyCF(BaseCF):
+    """Vocabulary custom field.
+
+    Supporting common vocabulary structure.
+    """
+
+    field_keys = ["id", "props", "title", "icon"]
+    """Return field's keys for querying.
+
+    These keys are used to select which information to return from the
+    vocabulary that is queried.
+    """
+
+    def __init__(self, name, vocabulary_id, multiple=False, dump_options=True):
+        """Constructor."""
+        super().__init__(name)
+        self.relation_cls = PIDRelation if not multiple else PIDListRelation
+        self.vocabulary_id = vocabulary_id
+        self.dump_options = dump_options
+        self.multiple = multiple
+
+    @property
+    def mapping(self):
+        """Return the mapping."""
+        _mapping = {
+            "type": "object",
+            "properties": {
+                "@v": {"type": "keyword"},
+                "id": {"type": "keyword"},
+                "title": {"type": "object", "dynamic": True},
```

Comment:
I guess this is why we cannot have strict schemas...

Reply:
To be honest, since we are setting `dynamic` directly on the inner field `title` I think we should be able to have a global `strict True` for the mapping. I've added a card in the board.